### PR TITLE
Travis-CI Build/Test integration into etest.

### PR DESCRIPTION
**Pull request by marshallmcmullen**
Added Travis-CI automatic build and test integration. Also removed a bunch of SF specific code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+[![Build Status](https://travis-ci.org/marshallmcmullen/ebash.svg?branch=master)](https://travis-ci.org/marshallmcmullen/ebash)
```
Comment (marshallmcmullen) [inline README.md]: This is nifty b/c it will show the build status at the top of the README in GitHub repo.

File: share/chroot.sh [hunk 2]
```diff
@@ -281,7 +281,7 @@ chroot_uninstall_filter()
 
 chroot_apt_setup()
 {
-    $(opt_parse CHROOT UBUNTU_RELEASE RELEASE HOST UBUNTU_ARCH)
```
Comment (marshallmcmullen) [inline share/chroot.sh]: This stuff with `RELEASE` and `HOST` are SF specific so I removed them.

File: share/efetch.sh [hunk 3]
```diff
@@ -105,7 +105,12 @@ __efetch_load_info()
         destination="${TMPDIR:-/tmp}"
     else
         destination=${urls[-1]}
-        unset urls[-1]
+        
+        if [[ ${BASH_VERSION} == 4.2* ]]; then
+            unset urls[${#urls[@]}-1]
```
Comment (marshallmcmullen) [inline share/efetch.sh]: The above fails on bash 4.2

File: unittest/ebash.etest [hunk 4]
```diff
@@ -100,15 +100,17 @@ EOF
 
 ETEST_ebash_bin_no_args()
 {
-    local output=$(EMSG_PREFIX="" EFUNCS_COLOR=1 ewarn hi |& ${EBASH_HOME}/bin/noansi)
+    local output
+    output=$(EMSG_PREFIX="" EFUNCS_COLOR=1 ewarn hi |& ${EBASH_HOME}/bin/noansi)
```
Comment (marshallmcmullen) [inline unittest/ebash.etest]: bashlint detects this as an error now.

File: unittest/efetch.etest [hunk 5]
```diff
@@ -7,6 +7,11 @@
 # as published by the Apache Software Foundation, either version 2 of the License, or (at your option) any later
 # version.
 
+# Test-only PGP files
```
Comment (marshallmcmullen) [inline unittest/efetch.etest]: We used to fetch bdr-jenkins files, now I have checked in static files.

File: share/efetch.sh (hunk 3 above)
Comment (ericpe) [reply share/efetch.sh]: This should be futureproofed like this:
`   if [[ ${BASH_VERSINFO[0]} -gt 4 || ( ${BASH_VERSINFO[0] -eq 4 && ${BASH_VERSINFO[1] -gt 1 ) ]] ; then`

Comment (ericpe) [reply share/efetch.sh]: well... unless it _only_ fails on 4.2 and works on 4.1 and 4.3

Comment (marshallmcmullen) [reply share/efetch.sh]: It fails on anything <= 4.2. So yeah, I should make this check better.

Comment (marshallmcmullen) [reply share/efetch.sh]: I fixed the above check to always use the safer backwards compatible version instead of the newer one.

**Pull request merged**